### PR TITLE
refactor: Remove SubscriptionManagerRunner

### PR DESCRIPTION
**Pull request by cindyyan317**

### CODE REVIEW

File: src/app/ClioApplication.cpp [hunk 1]
```diff
@@ -101,9 +102,10 @@ ClioApplication::run()
     auto backend = data::make_Backend(config_);
 
     // Manages clients subscribed to streams
-    auto subscriptionsRunner = feed::SubscriptionManagerRunner(config_, backend);
-
-    auto const subscriptions = subscriptionsRunner.getManager();
+    auto const workersNum = config_.valueOr<std::uint64_t>("subscription_workers", 1);
+    LOG(util::LogService::info()) << "Starting subscription manager with " << workersNum << " workers";
+    auto subscriptions =
+        std::make_shared<feed::SubscriptionManager>(util::async::PoolExecutionContext(workersNum), backend);
```
Comment (kuznetsss) [inline src/app/ClioApplication.cpp]: Maybe better to create a function `make_SubscriptionManager(config)` in `SubscriptionManager.hpp`?
Or a `SubscriptionManager`'s constructor taking config and backend?

File: src/feed/SubscriptionManager.hpp [hunk 2]
```diff
@@ -74,9 +72,9 @@ class SubscriptionManager : public SubscriptionManagerInterface {
      * @param backend The backend to use
      */
     template <class ExecutorCtx>
```
Comment (kuznetsss) [inline src/feed/SubscriptionManager.hpp]: Do we still need a template here? I think there is `AnyExecutionContext`.

File: src/app/ClioApplication.cpp (hunk 1 above)
Comment (cindyyan317) [reply src/app/ClioApplication.cpp]: The execution ctx is one of the constructor parameter. The number of threads is needed for only pool ctx. So logically, the cfg should not be sent to the constructor of subscription manager. Maybe better create a function. But creating this function will include the config in SubcriptionManager.hpp.

Comment (kuznetsss) [reply src/app/ClioApplication.cpp]: I was thinking that knowledge of which config option to take is related to `SubscriptionManagaer` so probably better to have all the information related to it in its own files. Including config shouldn't do any harm.

**Review by kuznetsss — APPROVED** [review]
LGTM 👍

**Pull request merged**